api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -120,19 +132,48 @@ emitLogRecord Logger {..} LogRecordArguments {..} = do
     SpanContext {traceId, spanId, traceFlags} <- getSpanContext currentSpan
     pure (traceId, spanId, traceFlags)
 
+  let logRecordAttributes =
+        LA.addAttributes
+          (loggerProviderAttributeLimits loggerProvider)
+          LA.emptyAttributes
+          attributes
+
+  when (LA.attributesDropped logRecordAttributes > 0) $ void logDroppedAttributes
+
   pure
     LogRecord
       { logRecordTimestamp = timestamp
       , logRecordObservedTimestamp
       , logRecordTracingDetails
-      , logRecordSeverityNumber = fmap mkSeverityNumber severityNumber
-      , logRecordSeverityText = severityText <|> (shortName . mkSeverityNumber =<< severityNumber)
+      , logRecordSeverityNumber = severityNumber
+      , logRecordSeverityText = severityText <|> (toShortName =<< severityNumber)
       , logRecordBody = body
       , logRecordResource = loggerProviderResource loggerProvider
       , logRecordInstrumentationScope = loggerInstrumentationScope
-      , logRecordAttributes =
-          addAttributes
-            (loggerProviderAttributeLimits loggerProvider)
-            emptyAttributes
-            attributes
+      , logRecordAttributes
+      }
+
+
+-- | WARNING: this function should only be used to emit logs from the hs-opentelemetry-api library. DO NOT USE this function in any other context.
```

Comment:
I'm not sure if this is the best way to communicate this. It would be ideal if these functions could be in an internal module that is not exported. That is not simple because it introduces a cyclic module dependency.

Reply:
Tracing/Core.hs uses it

Reply:
Fixed so that there is `OpenTelemetry.Internal.Logging.Core`